Test/c++/LIMITATIONS.md

```diff
@@ -0,0 +1,18 @@
+The C++ backend was written assuming that it would primarily support writing
+C/C++ style code in Dafny, which leads to some limitations in the current
+implementation.
```

Comment:
This file is better placed in the docs/Compilation folder.

Reply:
Ah, yes, that's a much better place.